### ISSUE TITLE
Allow tags to be added after a registration is created

### ISSUE THREAD

**Issue by jlcohoon** [CenterForOpenScience/osf.io#2954]
This comes from a community issue: https://github.com/CenterForOpenScience/community/issues/16
<h4>Current Behavior</h4>
Registrations do not allow the user to add or edit tags. If the project that was registered did not have any tags, the tag widget does not appear on the registration (for both contributors and non-contributors).

<h4>Proposed Behavior</h4>
Registrations should allow users to add and edit tags. If the project did not have tags at the time of registration, the tag widget should still be visible on the registration to contributors. Non-contributors should not see the tag widget at any time if there are no tags added. The alert to users describing what is registered should emphasize that tags can be changed (in addition to Mendeley and Zotero not being registered).

**Comment by chennan47** [comment]
What about ask user to add the tags they want before they submit the registration?

**Comment by brianjgeiger** [comment]
I think we're looking into adding a separate class of tags for people who aren't contributors for a project to be able to tag things. I don't expect there would be a problem tagging registrations with those tags, as they'd be more metadata than data. Would that be enough, or would you want contributors to be able to modify the user tags as well? The community-added tags would, I expect, be available through search, but in a way that makes them distinct from the contributor-added tags.

Also, I expect I'm going to spend time this epic on revising the Registration process in such a way that will make a clear what your registration is going to look like and let you spruce things up before submitting it. We might could have some "best practices for registrations" sorts of things to get people to consider whether they want tags.

**Comment by jlcohoon** [comment]
I believe that the intent with this, was for contributors on a project to later want their work to be more discoverable and to add tags to the project to help with that. So, contributors being able to modify user tags would be useful. It might be that they could only be added, not deleted. 

@saradbowman, you originally logged this issue, what do you think?

**Comment by lbanner** [comment]
@brianjgeiger is correct on Tagging, in multiple fashions, as probable project for this Epic .... will tag this issue as "Feature" and "Discuss"

**Comment by saradbowman** [comment]
Yes, original intent/request was for contributors on a project to make work more discoverable. I don't know what makes more sense, in terms of only adding and not deleting, etc., but my hunch is you should be able to do both.

**Comment by Ghalko** [comment]
#1332 - related
